### ISSUE TITLE
Reconstruct portion of a scan

### ISSUE THREAD

**Issue by pierrethibault** [ptycho/ptypy#116]
Implement a filter in PtyScan or PtydScan to allow users to select a subset of a scan. The question was raised in #109 (with a workaround by toggling `POD.active`).

**Comment by bjoernenders** [comment]
Yeah I also dream of the "searchlight" algorithm, that uses ML and DM similar to EPIE, only that you pick subsets and have a beam of activity wander through your dataset. The beam size matches your memory requirements. It requires however a certain selectivity on the data_package creation currently not in place.

**Comment by aaron-parsons** [comment]
Could we make use of the scan sharing mechanism (currently broken) to do this management? I'm looking at this as a better way to use the MPI for the CUDA stuff. We could adapt it to take a synchronous or asynchronous flag. You could imagine managing this with a class that is overridden for true scan sharing, or implemented with asynch, synch flags for "hidden" scan sharing, which could look like the search light.

**Comment by bjoernenders** [comment]
Should be addressed in the review of PtyScan #438

**Comment by daurer** [comment]
this already exists in the form of a bounding_box in the [HDF5Loader](https://github.com/ptycho/ptypy/blob/master/ptypy/experiment/hdf5_loader.py)
